plugins/modules/nd_site.py

```diff
@@ -0,0 +1,282 @@
+  register: query_result
+
+- name: Query all sites
+  cisco.nd.nd_site:
+    url: nd_host
+    username: admin
+    password: SomeSecretPassword
+    state: query
+  delegate_to: localhost
+  register: query_result
+"""
+
+RETURN = r"""
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec
+from ansible_collections.cisco.nd.plugins.module_utils.constants import SITE_TYPE_MAP
+
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(
+        site_password=dict(type="str", no_log=True),
+        site_username=dict(type="str"),
+        login_domain=dict(type="str"),
+        inband_epg=dict(type="str"),
+        site_name=dict(type="str", aliases=["name", "fabric_name"]),
+        url=dict(type="str", aliases=["host_name", "ip_address"]),
+        site_type=dict(type="str", choices=list(SITE_TYPE_MAP.keys())),
+        security_domains=dict(type="list", elements="str"),
+        latitude=dict(type="float"),
+        longitude=dict(type="float"),
+        re_register=dict(type="bool", default=False),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["site_name"]],
+            ["state", "present", ["site_name", "site_username", "site_password", "url", "site_type"]],
+        ],
+    )
+
+    nd = NDModule(module)
+
+    site_username = nd.params.get("site_username")
+
+    site_password = nd.params.get("site_password")
+    if site_password is not None:
+        # Make password base64 encoded
+        site_password = (base64.b64encode(str.encode(site_password))).decode("utf-8")
+
+    inband_epg = nd.set_to_empty_string_when_none(nd.params.get("inband_epg"))
+    if inband_epg is not "":
+        inband_epg = "uni/tn-mgmt/mgmtp-default/inb-{0}".format(inband_epg)
+
+    login_domain = nd.set_to_empty_string_when_none(nd.params.get("login_domain"))
+    site_name = nd.params.get("site_name")
+    url = nd.params.get("url")
+    site_type = nd.params.get("site_type")
+    latitude = nd.set_to_empty_string_when_none(nd.params.get("latitude"))
+    longitude = nd.set_to_empty_string_when_none(nd.params.get("longitude"))
+    security_domains = nd.params.get("security_domains")
+    re_register = nd.params.get("re_register")
+    state = nd.params.get("state")
+
+    path = "/nexus/api/sitemanagement/v4/sites"
+
+    if site_name:
+        site_info = next((site_dict.get("spec").get("name") for site_dict in nd.query_obj(path).get("items") if site_dict.get("spec").get("name") == site_name), None)
+        if site_info:
+            if re_register:
+                nd.request(path, method="DELETE")
+            else: 
+                path = "{0}/{1}".format(path, site_name)
+                nd.existing = nd.query_obj(path)
```

Comment:
Do we need the double call ( nd.query_obj() ) to retrieve more detailed information for the site or can we re-use the first call?